packages/smooth_app/lib/pages/scan/inherited_data_manager.dart

```diff
@@ -0,0 +1,58 @@
+import 'package:flutter/material.dart';
+
+class InheritedDataManager extends StatefulWidget {
+  const InheritedDataManager({
+    Key? key,
+    required this.child,
+  }) : super(key: key);
+
+  final Widget child;
+
+  static InheritedDataManagerState of(BuildContext context) {
+    return context
+        .dependOnInheritedWidgetOfExactType<_InheritedDataManagerProvider>()!
+        .data;
+  }
+
+  @override
+  State<InheritedDataManager> createState() => InheritedDataManagerState();
+}
+
+class InheritedDataManagerState extends State<InheritedDataManager> {
+  late bool showSearchCard;
+
+  @override
+  void initState() {
+    showSearchCard = false;
+    super.initState();
+  }
+
+  void resetShowSearchCard(bool newValue) {
+    setState(() {
+      showSearchCard = newValue;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return _InheritedDataManagerProvider(
+      data: this,
+      child: widget.child,
+    );
+  }
+}
+
+class _InheritedDataManagerProvider extends InheritedWidget {
```

Comment:
We are using Provider for state managment, what about using it to keep the code consistent

Reply:
I need to pass a flag from page_manager to scan_carousel, InheritedWidget might be a better approach to prevent adding params to constructors and rebuilding the whole tree